### PR TITLE
Manual filter creation

### PR DESCRIPTION
**Pull request by YenguiSeddik**

### CODE REVIEW

File: src/main/java/org/gridsuite/filter/server/dto/ManualFilter.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        this.filterEquipmentsAttributes = filterEquipmentsAttributes;
+    }
+
+    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
+    @Override
+    public FilterType getType() {
+        return FilterType.MANUAL;
+    }
+
+    public EquipmentType getEquipmentType() {
```
Comment (sBouzols) [inline src/main/java/org/gridsuite/filter/server/dto/ManualFilter.java:52]: If you have the @Getter on the class definition why do you need to implement this one ?

File: src/main/java/org/gridsuite/filter/server/dto/FormFilter.java [hunk 2]
```diff
@@ -38,7 +38,7 @@ public FormFilter(UUID id, Date creationDate, Date modificationDate, AbstractEqu
     @JsonProperty(access = JsonProperty.Access.READ_ONLY)
     @Override
     public FilterType getType() {
-        return FilterType.FORM;
+        return FilterType.AUTOMATIC;
     }
 
     public AbstractEquipmentFilterForm getEquipmentFilterForm() {
```
Comment (sBouzols) [inline src/main/java/org/gridsuite/filter/server/dto/FormFilter.java:44]: I see it's the same here ... Do we need this getter implementation ?

File: src/main/java/org/gridsuite/filter/server/dto/ManualFilter.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+ * @author Seddik Yengui <seddik.yengui at rte-france.com>
+ */
+
+@Getter
+@Schema(description = "Manual Filters", allOf = AbstractFilter.class)
+@SuperBuilder
+@NoArgsConstructor
+public class ManualFilter extends AbstractFilter {
+
+    EquipmentType equipmentType;
```
Comment (sBouzols) [inline src/main/java/org/gridsuite/filter/server/dto/ManualFilter.java:33]: should be private ?

**Review by sBouzols — CHANGES_REQUESTED** [review]
remove the empty derby.log file from this PR ?
Some comments from code review.
Will test

**Review by etiennehomer — APPROVED** [review]

File: src/main/java/org/gridsuite/filter/server/dto/FormFilter.java (hunk 2 above)
Comment (etiennehomer) [reply src/main/java/org/gridsuite/filter/server/dto/FormFilter.java:44]: We don't have a type attribute in this class, we can't do it with the @Getter

**Review by sBouzols — APPROVED** [review]
Code Review OK
Tests OK

File: src/main/java/org/gridsuite/filter/server/dto/AbstractFilter.java [hunk 4]
```diff
@@ -26,7 +26,8 @@
 )
 @JsonSubTypes({//Below, we define the names and the binding classes.
     @JsonSubTypes.Type(value = ScriptFilter.class, name = "SCRIPT"),
-    @JsonSubTypes.Type(value = FormFilter.class, name = "FORM"),
+    @JsonSubTypes.Type(value = FormFilter.class, name = "AUTOMATIC"),
```
[the remark this answers is not in the document]
Comment (YenguiSeddik) [reply src/main/java/org/gridsuite/filter/server/dto/AbstractFilter.java]: Ok, done

**Pull request merged**